### ISSUE TITLE
[RISCV64] Return Fetching PC + 2 Instead of RA

### ISSUE THREAD

**Issue by hughes-c** [sstsimulator/sst-elements#1767]
Fetching incorrect instruction after return -- PC + 2 instead of loading RA:
10552 - > 10baa -> 10bac -> 10bae -> 10bb0

Should be: 
10552 - > 10baa -> 10bac -> 10bae -> 10556

```
   10552:	658000ef          	jal	ra,10baa <__set_thread_area>
   10556:	02054f63          	bltz	a0,10594 <__init_tp+0x50>

0000000000010baa <__set_thread_area>:
   10baa:	822a                	mv	tp,a0
   10bac:	4501                	li	a0,0
   10bae:	8082                	ret

0000000000010bb0 <.rodata>:
	...
   10bb8:	7665642f          	0x7665642f
   10bbc:	6c756e2f          	0x6c756e2f
   10bc0:	006c                	addi	a1,sp,12
```

[Core:    0/12174780]: [fill-decode]: ins-addr: 0x10bae line-offset: 46 line-start=68480 / 0x10b80
[Core:    0/12174780]: [fill-decode]: load 4 bytes from this line.
[Core:    0/12174780]: [fill-decode]: line-filled: yes
[Core:    0/12174780]: ---> performing a decode for ip=0x10bae
[Core:    0/12174780]: [decode] -> addr: 0x10bae / ins: 0x00008082
[Core:    0/12174780]: [decode] -> ins-bytes: 0x00008082
[Core:    0/12174780]: -----> RVC op_code: 2
[Core:    0/12174780]: ---> RVC function code = 32768 / 0x8000
[Core:    0/12174780]: ------> RVC 12b: 0 / rs1=1 / rs2=0
[Core:    0/12174780]: --------> (generates) RVC JR 1
[Core:    0/12174780]: [decode] -> 16bit RVC format / ins-op-code-family: 2 / 0x2
[Core:    0/12174780]: ---> bundle generates 1 micro-ops
[Core:    0/12174780]: ---> cycle is completed, ip=0x10bae
[Core:    0/12174780]: -> Decode step for thr: 0
[Core:    0/12174780]: ---> Max decodes per cycle: 2
[Core:    0/121747FATAL: [Core:    0/12174780]: STOP
80]: ---> Found uop bundle for ip=0x10bae, loading from cache...
[Core:    0/12174780]: ----> Bundle contains 1 entries.
[Core:    0/12174780]: ----> contains a branch: 0x10bae / predicted (not-found in predictor): 0x10bb2, pc-increment: 2
[Core:    0/12174780]: ----> branch? yes, ip=0x10bb0 + inc=0 = new-ip=0x10bb0
[Core:    0/12174780]: ---> uop not found, but is located in the predecode i0-icache (ip=0x10bb0)

**Comment by nmhamster** [comment]
In the absence of correctly predicted branches, the decoder will keep processing. This will generate a sequence of micro-ops which will never be executed because the retirement engine will identify the correct branching targets and clear the pipeline performing a restart. Due to the development nature of the RISCV decoder, the decode pipeline is currently set to stall on detection of zero micro-op bundles at decode completion. Long term correct behavior for Vanadis can be seen in the MIPSO32 decoder, which advances and produces decode-fault micro-ops which do not cause pipeline errors until the engine attempts retirement.

**Comment by nmhamster** [comment]
Closing because functionality for handling STOP conditions is now merged (in PR #1768). Reopen if needed.
